roles/roles.go

```diff
@@ -0,0 +1,58 @@
+package roles
+
+import (
+	"slices"
+
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/services"
+)
+
+// Role is the string literal of a role.
+type Role string
+
+// RoleRegistration is a struct that contains the display name and roles for a service.
+type RoleRegistration struct {
+	DisplayName string
+	Roles       []Role
+}
+
+var (
+	// services.Dotcom
+	dotcomDisplayName string = "Sourcegraph Dotcom"
+	dotcomRoles              = []Role{
+		RoleDotcomSiteadmin,
+	}
+)
+
+const (
+	// Roles for services.Dotcom
+
+	// RoleDotcomSiteadmin is the role for site admins on dotcom
+	RoleDotcomSiteadmin Role = "site_admin"
+)
+
+// Registered returns a map of all registered roles for all services.
+type RegisteredRoles map[services.Service]RoleRegistration
+
+func Registered() RegisteredRoles {
```

Comment:
IMO we should have 2 explicit things:

1. A helper that surfaces the Entitle version of the roles set (`[]EntitleRoleRegistration`)
2. A helper that surfaces `[]Role` with helpers on `type Roles []Role` like `(Roles).Contains(service, role)` or similar